var/spack/repos/builtin/packages/survey/package.py

```diff
@@ -0,0 +1,210 @@
+            description="Build mpi collector for mvapich2\
+                         MPI when variant is enabled.")
+    variant('mvapich', default=False,
+            description="Build mpi collector for mvapich\
+                         MPI when variant is enabled.")
+    variant('mpich2', default=False,
+            description="Build mpi collector for mpich2\
+                         MPI when variant is enabled.")
+    variant('mpich', default=False,
+            description="Build mpi collector for mpich\
+                         MPI when variant is enabled.")
+    variant('build_type', default='RelWithDebInfo',
+            description='The build type to build',
+            values=('Debug', 'Release', 'RelWithDebInfo'))
+
+    # must have cmake at 3.12 or greater to find python3
+    depends_on('cmake@3.12:')
+
+    # for collectors
+    depends_on("libmonitor@2021.04.27+commrank", type=('build', 'link', 'run'))
+
+    depends_on("papi@5:", type=('build', 'link', 'run'))
+    depends_on("gotcha@master", type=('build', 'link', 'run'))
+    depends_on("llvm-openmp@9.0.0", type=('build', 'link', 'run'))
+
+    # MPI Installations
+    depends_on("openmpi", when='+openmpi')
+    depends_on("mpich@:", when='+mpich')
+    depends_on("mpich@2:", when='+mpich2')
+    depends_on("mvapich2@2:", when='+mvapich2')
+    depends_on("mvapich2@:1", when='+mvapich')
+    depends_on("mpt", when='+mpt')
+
+    depends_on("python@3:", type=('build', 'link', 'run'))
+    depends_on("py-setuptools", type=('build', 'link', 'run'))
+    depends_on("py-pip", type=('build', 'link', 'run'))
+    depends_on("py-python-dateutil", type=('build', 'link', 'run'))
+    depends_on("py-pandas", type=('build', 'link', 'run'))
+    depends_on("py-psutil", type=('build', 'link', 'run'))
+    depends_on("py-sqlalchemy", type=('build', 'link', 'run'))
+    depends_on("py-pbr", type=('build', 'link', 'run'))
+
+    extends('python')
+
+    parallel = False
+
+    def set_mpi_cmake_options(self, spec, cmake_options):
+        # Appends to cmake_options the options that will enable the appropriate
+        # MPI implementations
+
+        mpi_options = []
+
+        # openmpi
+        if spec.satisfies('+openmpi'):
+            mpi_options.append('-DOPENMPI_DIR=%s' % spec['openmpi'].prefix)
+        # mpich
+        if spec.satisfies('+mpich'):
+            mpi_options.append('-DMPICH_DIR=%s' % spec['mpich'].prefix)
+        # mpich2
+        if spec.satisfies('+mpich2'):
+            mpi_options.append('-DMPICH2_DIR=%s' % spec['mpich2'].prefix)
+        # mvapich
+        if spec.satisfies('+mvapich'):
+            mpi_options.append('-DMVAPICH_DIR=%s' % spec['mvapich'].prefix)
+        # mvapich2
+        if spec.satisfies('+mvapich2'):
+            mpi_options.append('-DMVAPICH2_DIR=%s' % spec['mvapich2'].prefix)
+        # mpt
+        if spec.satisfies('+mpt'):
+            mpi_options.append('-DMPT_DIR=%s' % spec['mpt'].prefix)
+
+        cmake_options.extend(mpi_options)
```

Comment:
```suggestion
    def set_mpi_cmake_options(self, spec, cmake_options):
        # Appends to cmake_options the options that will enable the appropriate
        # MPI implementations
        mpi_options = ['-D%s_DIR=%s' % (spec['mpi'].name.upper(), spec['mpi'].prefix)]
        cmake_options.extend(mpi_options)
```
Once you make that change, it's easier to inline this back into the `cmake_args` method.

Reply:
@becker33  Hi Greg - Our tool builds multiple MPI collectors, so just going down to one "mpi" doesn't work for our tool. We can specify +openmpi+mpich and generate multiple collectors.